packages/go/analysis/ad/gpos.go

```diff
@@ -0,0 +1,255 @@
+// Copyright 2025 Specter Ops, Inc.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package ad
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+
+	"github.com/specterops/bloodhound/analysis"
+	"github.com/specterops/bloodhound/dawgs/graph"
+	"github.com/specterops/bloodhound/dawgs/ops"
+	"github.com/specterops/bloodhound/dawgs/query"
+	"github.com/specterops/bloodhound/dawgs/traversal"
+	"github.com/specterops/bloodhound/dawgs/util/channels"
+	"github.com/specterops/bloodhound/graphschema/ad"
+)
+
+func PostGPOs(ctx context.Context, db graph.Database) (*analysis.AtomicPostProcessingStats, error) {
+	if domainNodes, err := fetchCollectedDomainNodes(ctx, db); err != nil {
+		return &analysis.AtomicPostProcessingStats{}, err
+	} else {
+		operation := analysis.NewPostRelationshipOperation(ctx, db, "GPOs Post Processing")
+
+		for _, domain := range domainNodes {
+			if err := operation.Operation.SubmitReader(func(ctx context.Context, tx graph.Transaction, outC chan<- analysis.CreatePostRelationshipJob) error {
+				processGPOs(ctx, tx, outC, domain, []graph.ID{}, []graph.ID{}, []graph.ID{})
+				return nil
+			}); err != nil {
+				slog.ErrorContext(ctx, fmt.Sprintf("Failded processing GPOs of domain %d: %v", domain.ID, err))
+			}
+		}
+
+		return &operation.Stats, operation.Done()
+	}
+}
+
+func processGPOs(ctx context.Context, tx graph.Transaction, outC chan<- analysis.CreatePostRelationshipJob, node *graph.Node, gposInherited, gposInheritedEnforced, gpoAppliers []graph.ID) {
+	applyingGPOs := []graph.ID{}
+	gposInheritedEnforcedNew := gposInheritedEnforced
+	blocksGPOInheritance := false
+
+	if node.Kinds.ContainsOneOf(ad.Domain, ad.OU) {
+		// Find principals with permission to link GPOs to this node
+		if gpoAppliersOnThisContainer, err := fetchGPOAppliers(tx, node.ID); err != nil {
```

Comment:
Looks like these `slog.ErrorContext`s in this function have `Failded` instead of `Failed`